miden-crypto/src/dsa/ecdsa_secp256k1/mod.rs

```diff
@@ -0,0 +1,416 @@
+    elliptic_curve::rand_core::{CryptoRng, RngCore},
+};
+
+use crate::{
+    Felt, SequentialCommit, StarkField, Word,
+    utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable},
+};
+
+// CONSTANTS
+// ================================================================================================
+
+const SECRET_KEY_BYTES: usize = 32;
+const PUBLIC_KEY_BYTES: usize = 33; // we use the compressed format
+const SIGNATURE_BYTES: usize = 66;
+const SCALARS_SIZE_BYTES: usize = 32;
+
+// ECDSA HASHER
+// ================================================================================================
+
+#[repr(u8)]
+#[derive(Debug, Clone, Copy)]
+pub enum EcdsaHasher {
+    Sha256 = 0,
+    Keccak = 1,
+}
+
+impl EcdsaHasher {
+    /// Hashes a message using the variant of the ECDSA hasher.
+    fn hash(&self, message: &[u8]) -> [u8; 32] {
+        match self {
+            EcdsaHasher::Sha256 => {
+                use k256::sha2::{Digest, Sha256};
+                let mut hasher = Sha256::new();
+                hasher.update(message);
+                hasher.finalize().into()
+            },
+            EcdsaHasher::Keccak => {
+                use sha3::{Digest, Keccak256};
+                let mut hasher = Keccak256::new();
+                hasher.update(message);
+                hasher.finalize().into()
+            },
+        }
+    }
+
+    /// Converts the hasher variant to a `u8`.
+    pub fn to_byte(&self) -> u8 {
+        *self as u8
+    }
+
+    /// Converts a `u8` to its corresponding hasher variant.
+    pub fn from_byte(byte: u8) -> Option<Self> {
+        if byte <= 1 {
+            Some(unsafe { core::mem::transmute::<u8, EcdsaHasher>(byte) })
+        } else {
+            None
+        }
+    }
+}
+
+// SECRET KEY
+// ================================================================================================
+
+/// Secret key for ECDSA signature verification over secp256k1 curve.
+#[derive(Clone)]
+pub struct SecretKey {
+    inner: SigningKey,
+}
+
+impl SecretKey {
+    /// Generates a new random secret key using the OS random number generator.
+    #[cfg(feature = "std")]
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        let mut rng = k256::elliptic_curve::rand_core::OsRng;
+
+        Self::with_rng(&mut rng)
+    }
+
+    /// Generates a new secret key using the provided random number generator.
+    pub fn with_rng<R: CryptoRng + RngCore>(rng: &mut R) -> Self {
+        let signing_key = SigningKey::random(rng);
+        Self { inner: signing_key }
+    }
+
+    /// Gets the corresponding public key for this secret key.
+    pub fn public_key(&self) -> PublicKey {
+        let verifying_key = self.inner.verifying_key();
+        PublicKey { inner: *verifying_key }
+    }
+
+    /// Signs a message (represented as a Word) with this secret key.
+    pub fn sign(&mut self, message: Word, hasher: EcdsaHasher) -> Signature {
```

Comment:
I think this works. Another approach could be to have dedicated functions  - e.g., `sign_with_sha256()` or `sign_with_keccak()`, but I think the approach with the enum is a bit cleaner.